### PR TITLE
DOC: add manual example for causal consistency

### PR DESCRIPTION
**Pull request by prashantmital**
Closes [PYTHON-1623](https://jira.mongodb.org/browse/PYTHON-1623).

### CODE REVIEW

File: test/test_examples.py [hunk 1]
```diff
@@ -1114,5 +1117,37 @@ def shipment_transaction(session):
...
+        # Causal consistency examples
+        client = self.client
+        self.addCleanup(client.drop_database, 'test')
+        client.test.drop_collection('items')
+        client.test.items.insert_one({'sku': "111", 'name': 'Peanuts', 'start':datetime.datetime.today()})
+
+        # Start Causal Consistency Example 1
+        with client.start_session(causal_consistency=True) as s1:
+            current_date = datetime.datetime.today()
+            items = client.get_database('test', write_concern=WriteConcern(w='majority')).items
```
Comment (ShaneHarvey) [inline test/test_examples.py]: I don't think this example needs the majority write concern.

File: test/test_examples.py [hunk 2]
```diff
@@ -1114,5 +1117,37 @@ def shipment_transaction(session):
...
 
 
+class TestCausalConsistencyExamples(IntegrationTest):
+    @client_context.require_version_min(3, 6, 0)
+    def test_causal_consistency(self):
+        # Causal consistency examples
+        client = self.client
+        self.addCleanup(client.drop_database, 'test')
+        client.test.drop_collection('items')
+        client.test.items.insert_one({'sku': "111", 'name': 'Peanuts', 'start':datetime.datetime.today()})
```
Comment (ShaneHarvey) [inline test/test_examples.py]: Wrap at 80 chars.

File: test/test_examples.py [hunk 3]
```diff
@@ -1114,5 +1117,37 @@ def shipment_transaction(session):
...
+                {'sku': "nuts-111", 'name': "Pecans",
+                 'start': current_date}, session=s1)
+        # End Causal Consistency Example 1
+
+        # Start Causal Consistency Example 2
+        with client.start_session(causal_consistency=True) as s2:
+            s2.advance_cluster_time(s1.cluster_time)
+            s2.advance_operation_time(s1.operation_time)
+
+            items = client.get_database('test', read_preference=ReadPreference.SECONDARY).items
```
Comment (ShaneHarvey) [inline test/test_examples.py]: Wrap at 80 chars.

File: test/test_examples.py [hunk 4]
```diff
@@ -1114,5 +1117,37 @@ def shipment_transaction(session):
...
+                {'$set': {'end': current_date}}, session=s1)
+            items.insert_one(
+                {'sku': "nuts-111", 'name': "Pecans",
+                 'start': current_date}, session=s1)
+        # End Causal Consistency Example 1
+
+        # Start Causal Consistency Example 2
+        with client.start_session(causal_consistency=True) as s2:
+            s2.advance_cluster_time(s1.cluster_time)
+            s2.advance_operation_time(s1.operation_time)
```
Comment (ShaneHarvey) [inline test/test_examples.py]: I really wish this was easier, like:
```
s2.causally_consitent_with(s1)
```

**Comment by prashantmital** [comment]
@ShaneHarvey I incorporated your suggested changes.

**Pull request merged**